google-cloud-storage/lib/google/cloud/storage/service.rb

```diff
@@ -460,6 +472,8 @@ def compose_file bucket_name,
           source_objects = compose_file_source_objects source_files, if_source_generation_match
           compose_req = Google::Apis::StorageV1::ComposeRequest.new source_objects: source_objects,
                                                                     destination: destination_gapi
+          is_idempotent = Retry.retry?(if_generation_match: if_generation_match)
+          options = is_idempotent ? key_options(key) : key_options(key).merge(retries: 0)
```

Comment:
Same here, compose also requires preconditions for the destination specifically in order to allow retries.

Reply:
Same goes here. `if_source_generation_match` is for source files and `if_generation_match` is for destinations files.
@dazuma @quartzmo can you please confirm if you have any idea?

Reply:
I believe that's correct.